app/services/imports/source_detector.rb

```
@@ -0,0 +1,235 @@
+      begin
+        File.open(file_path, 'rb') do |f|
+          partial_content = f.read(4096) # Try a bit more content
+          Oj.load(partial_content, mode: :compat)
+        end
+      rescue Oj::ParseError, JSON::ParserError
+        nil
+      end
+    else
+      nil
+    end
+  end
```

RuboCop (1.76.1)

[convention] Method has too many lines. [18/10]

(Metrics/MethodLength)